src/models/NetworkAppController.js

```diff
@@ -161,9 +168,69 @@ export default class NetworkAppController {
     }
 
     // TODO: Check that package version matches the requested one
+    // TODO: Do not invoke setStdlib if matches existing one
     log.info(`Connecting to public deployment of ${this.package.stdlib.name} in ${this.network}`);
     const stdlibAddress = StdlibProvider.from(this.package.stdlib.name, this.network);
     await this.appManagerWrapper.setStdlib(stdlibAddress);
     this.networkPackage.stdlib = { address: stdlibAddress, ... this.package.stdlib };
   }
+
+  checkContracts(throwIfFail = false) {
```

Comment:
I would try to be more explicit here, it is not clear what we are checking to me. Maybe we can rename this method to `checkDeployedContractsHaveNotChanged` or so